### PR TITLE
feat: add docker hot reload for frontend

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Jira: [FSADT2-670](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-670)

Add front end hot reload function to docker

### How to test
1. Run with docker compose command
    I prefer `docker compose --env-file .env up`, the `.env` file has `FORESTCLIENTAPI_KEY`
    Alternatively you can export this env var to your terminal
2. Once the front end container is running, change some text like you would doing front end development, the page should be hot reloaded reflecting the change.
3. Note that the first re-compile is quite slow(at least on my machine), but the subsequent ones are fast 

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
local

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-141-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-141-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-141-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-141-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-141-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-141-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

**Review by DerekRoberts — APPROVED** [review]
Idea - add a non-linked volume to store node_modules.  ...Nice addition.  :)
```
volumes: ... -/app/node_modules
```

File: frontend/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Build
```
[the remark this answers is not in the document]
Comment (DerekRoberts) [reply frontend/Dockerfile.dev:1]: Healthchecks are ignored by OpenShift/Kubernetes.  Consider this optoinal and feel free to dismiss.

[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/Dockerfile.dev:1]: Oh, a check is definitely great to have, let me steal some code

[the remark this answers is not in the document]
Comment (craigyu) [reply frontend/Dockerfile.dev:1]: @DerekRoberts added user, expose and healthcheck

**Pull request merged**